src/main/kotlin/io/getunleash/UnleashClient.kt

```diff
@@ -98,13 +98,16 @@ class UnleashClient(
     }
 
     override fun updateContext(context: UnleashContext): CompletableFuture<Void> {
-        refreshPolicy.context = context
         this.unleashContext = context
+        refreshPolicy.context = this.getContext()
         return refreshPolicy.refreshAsync()
     }
 
     override fun getContext(): UnleashContext {
-        return unleashContext
+        return unleashContext.copy(
+                appName = unleashContext.appName ?: unleashConfig.appName,
+                environment = unleashContext.environment ?: unleashConfig.environment
+        )
     }
```

Comment:
This will create a brand new class every time we call it, instead of using the existing on the class. I'm wondering if we should rather do the fallback to config values on constructing the client.

Reply:
I was trying to cater for a use case when people overwrite appName with updateContext in the first call and then another updateContext removes appName and then we should fall back to the config appName. So it gives you a fallback no matter how you used updateContext in the past.

Reply:
In other frontend SDKs updateContext can't update appName so we can save appName from config during construction. Since this SDK allows for appName overwrites from context it complicates the solution space a bit.

Reply:
Yeah, true. Then I think I'm fine with this. Let's merge.